### PR TITLE
FOGL-6740 Add support for IsNull and NotNull conditions

### PR DESCRIPTION
**Pull request by MarkRiddoch**
Signed-off-by: Mark Riddoch <mark@dianomic.com>

### CODE REVIEW

**Review by pintomax — APPROVED** [review]

**Comment by ashish-jabble** [comment]
@MarkRiddoch @praveen-garg I have added Payload builder changes including unit tests in this branch. Also I ran the custom CI JOB to check the health system tests for regression check.
Kindly review.

**Comment by ashish-jabble** [comment]
fyi, System health test run okay with the given change. I have checked on both ub18 and ub20 platform.

**Comment by MarkRiddoch** [comment]
Hi @ashish-jabble Changes look good to me. I will not be around tomorrow, so please merge this as soon as you are happy and the review has been passed.

File: python/fledge/common/storage_client/payload_builder.py [hunk 1]
```diff
@@ -60,6 +60,12 @@ def verify_condition(arg):
                 if arg[1] in ['in', 'not in']:
                     if isinstance(arg[2], list):
                         retval = True
+            elif len(arg) == 2:
+                # There is no 3rd arg required for below operations
+                if arg[1] in ['isnull', 'notnull']:
+                    retval = True
+            else:
+                retval = False
```
Comment (praveen-garg) [inline python/fledge/common/storage_client/payload_builder.py:68]: This else is redundant.

File: tests/unit/python/fledge/common/storage_client/test_payload_builder.py [hunk 2]
```diff
@@ -95,35 +95,44 @@ def test_conditions_payload(self, test_input, expected):
...
+        (["plugin", "=", "sinusoid"], ["deprecated_ts", "notnull"], _payload("data/payload_and_where_notnull.json"))
     ])
     def test_and_where_payload(self, test_input_1, test_input_2, expected):
         res = PayloadBuilder().WHERE(test_input_1).AND_WHERE(test_input_2).payload()
         assert expected == json.loads(res)
 
     @pytest.mark.parametrize("test_input_1, test_input_2, test_input_3, expected", [
-        (["name", "=", "test"], ["id", ">", 3], ["value", "!=", 0], _payload("data/payload_and_where2.json"))
+        (["name", "=", "test"], ["id", ">", 3], ["value", "!=", 0], _payload("data/payload_and_where2.json")),
+        (["plugin", "=", "sinusoid"], ["deprecated_ts", "notnull"], ["event", "=", "Ingest"], _payload("data/payload_multiple_and_where.json"))
```
Comment (praveen-garg) [inline tests/unit/python/fledge/common/storage_client/test_payload_builder.py:117]: better to add for isnull as well

File: tests/unit/python/fledge/common/storage_client/test_payload_builder.py [hunk 3]
```diff
@@ -95,35 +95,44 @@ def test_conditions_payload(self, test_input, expected):
...
+        (["event", "=", "Egress"], ["deprecated_ts", "notnull"], _payload("data/payload_or_where_notnull.json"))
     ])
     def test_or_where_payload(self, test_input_1, test_input_2, expected):
         res = PayloadBuilder().WHERE(test_input_1).OR_WHERE(test_input_2).payload()
         assert expected == json.loads(res)
 
     @pytest.mark.parametrize("test_input_1, test_input_2, test_input_3, expected", [
-        (["name", "=", "test"], ["id", ">", 3], ["value", "!=", 0], _payload("data/payload_or_where2.json"))
+        (["name", "=", "test"], ["id", ">", 3], ["value", "!=", 0], _payload("data/payload_or_where2.json")),
+        (["plugin", "=", "http_north"], ["deprecated_ts", "isnull"], ["event", "=", "Egress"],
```
Comment (praveen-garg) [inline tests/unit/python/fledge/common/storage_client/test_payload_builder.py:134]: better to add for notnull as well

**Review by praveen-garg — APPROVED** [review]
LGTM!

**Comment by MarkRiddoch** [comment]
Merging this so that we can free the GUI to continue. I suggest we raise a Jira to resolve the tests that Praveen suggested.

**Pull request merged**